pyvrp/read.py

```diff
@@ -104,16 +104,21 @@ def read(
     else:
         coords = np.zeros((num_clients, 2), dtype=int)
 
-    if "time_window" in instance:
-        time_windows = round_func(instance["time_window"])
-    else:
-        time_windows = np.repeat([[0, _INT_MAX]], num_clients, axis=0)
-
     if "service_time" in instance:
         service_times = round_func(instance["service_time"])
     else:
         service_times = np.zeros(num_clients, dtype=int)
 
+    if "time_window" in instance:
+        time_windows = round_func(instance["time_window"])
+    else:
+        # The default value for the latest time window based on the maximum
+        # route duration. This ensures that the time window constraints are
+        # always satisfied.
+        bound = num_clients * (edge_weight.max() + service_times.max())
+        bound = bound if bound < _INT_MAX else _INT_MAX
+        time_windows = np.repeat([[0, bound]], num_clients, axis=0)
```

Comment:
So this solves our CVRP/VRPTW INT_MAX woes? Good!

Suggestion: `bound = <ternary expression>` is a somewhat non-obvious way of writing `bound = min(bound, _INT_MAX)`.